src/NServiceBus.Persistence.Sql.TransactionalSession/SqlPersistenceOpenSessionOptions.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace NServiceBus.TransactionalSession
+{
+    using System;
+    using System.Collections.Generic;
+    using Transport;
+
+    /// <summary>
+    ///
+    /// </summary>
+    public sealed class SqlPersistenceOpenSessionOptions : OpenSessionOptions
+    {
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="tenantIdHeaderName"></param>
+        /// <param name="tenantId"></param>
+        public SqlPersistenceOpenSessionOptions(string tenantIdHeaderName = null, string tenantId = null)
+        {
+            var headers = new Dictionary<string, string>();
+            if (tenantIdHeaderName != null)
+            {
+                if (string.IsNullOrEmpty(tenantId))
+                {
+                    throw new Exception("A tenant header is available, but the value is missing.");
+                }
+                headers.Add(tenantIdHeaderName, tenantId);
```

Comment:
isn't setting headers the responsibility of the sender?

Reply:
> the connection to the right DB is established for the outbox records

that should be done via the control message and that's set on the metadata though?

Reply:
The metadata is propagated to headers of the control message. We also need to add it to the headers of the IncomingMessage to make sure when the outbox interaction happens, the appropriate connection will be opened. Do you foresee a problem there? Otherwise, I'm not clear what you are hinting at. It might be though that I'm just not yet understanding where this is going.

Reply:
sorry I seem to have misread the headers part and the use of the extensions. Resolving this comment.

Reply:
No worries. It is good to double and tripple check these things